pkg/cloudprovider/gce/gce.go

```diff
@@ -466,10 +468,10 @@ func (gce *GCECloud) getInstanceByName(name string) (*compute.Instance, error) {
 }
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
-func (gce *GCECloud) NodeAddresses(instance string) ([]api.NodeAddress, error) {
-	externalIP, err := gce.getExternalIP(instance)
+func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
```

Comment:
Can you add a TODO to rename/redefine this function across the implementations of the interface to make it clear that it is getting the current instance's Addresses.